mantidproject/mantid
# Add signal-to-noise threshold to PDCalibration

**Description of work.**

The issue: https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=245



**To test:**



 


From mantid workbench, run a modified version of the script originally provided by [66j](https://code.ornl.gov/66j):
```
from mantid.simpleapi import *
LoadNexus(Filename='/SNS/SNAP/shared/Malcolm/Temp/calFile_8x8.nxs',OutputWorkspace='calFile_8x8')
peaksToFit = '2.0346,1.7620,1.2459,1.0625,1.0173,0.8810,0.8085,0.7880'
PDCalibration(InputWorkspace='calFile_8x8', TofBinning='2500,10,14500',\  PeakFunction='Gaussian',BackgroundType='Linear',\  PeakPositions=peaksToFit, CalibrationParameters='DIFC', OutputCalibrationTable='_cal',\  DiagnosticWorkspaces='_diag',MaxChiSq=7,PeakWindow=0.2,MinimumPeakHeight=20,MinimumSignalToNoiseRatio=0)
```
By changing the value of `MinimumSignalToNoiseRatio`, you can see the effect the new property has on the _cal_mask->ShowInstrument pixel map. Also, when `MinimumSignalToNoiseRatio` is > 0, the number of peaks rejected because of the low signal-to-noise ratio will be reported as a [Notice] in the log.









---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
